### PR TITLE
Added Boost to some BUILD files and stddef.h to Fix Compiler Errors

### PR DESCRIPTION
**Pull request by EvanMorcom**
For whatever reason I could not get master to build after the most recent changes.

Adding these libraries fixed it.

### Review Checklist

- [X] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [X] **Remove all commented out code**
- [X] **Remove extra print statements**: for example, those just used for testing
- [X] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [X] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/firmware_new/tools/communication/transfer_media/BUILD [hunk 1]
```diff
@@ -14,5 +14,7 @@ cc_library(
         ":transfer_medium",
         "//software/multithreading:thread_safe_buffer",
         "@g3log",
+        "@boost//:array",
+	"@boost//:asio",
```
Comment (MathewMacDougall) [inline src/firmware_new/tools/communication/transfer_media/BUILD]: nit: indenting not aligned (should be fixed by running formatting)

File: src/firmware/main/math/polynomial_1d.c [hunk 2]
```diff
@@ -1,6 +1,7 @@
 #include "firmware/main/math/polynomial_1d.h"
 
 #include <math.h>
+#include <stddef.h>
```
Comment (MathewMacDougall) [inline src/firmware/main/math/polynomial_1d.c]: @garethellis0 you mentioned in Slack you had an idea of the cause? (Seems to be the same failure/file as Haomiao). Any idea why adding the include to the header file doesn't work?

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]
minor stuff otherwise LGTM. @EvanMorcom weren't there also changes in the network_client?

File: src/firmware_new/tools/communication/transfer_media/BUILD (hunk 1 above)
Comment (EvanMorcom) [reply src/firmware_new/tools/communication/transfer_media/BUILD]: Ran formatting, should be fixed

File: src/firmware/main/math/polynomial_1d.c (hunk 2 above)
Comment (EvanMorcom) [reply src/firmware/main/math/polynomial_1d.c]: I moved the STDDEF.h file to the header and it works 👍

**Review by garethellis0 — APPROVED** [review]

**Review by jonathanlew — APPROVED** [review]
LGTM! :+1:

**Review by MathewMacDougall — APPROVED** [review]

**Review by HaomiaoZ — APPROVED** [review]

**Comment by MathewMacDougall** [comment]
@EvanMorcom :shipit: :shipit:

**Review by akhilveeraghanta — APPROVED** [review]
LGTM :shipit:

**Pull request merged**